File: ip-messaging/rest/messages/retrieve-messages/retrieve-messages.5.x.rb
require 'twilio-ruby'

# Get your Account SID, Auth Token, Service SID and Channel SID from twilio.com/console
account_sid = 'ACCOUNT_SID'
auth_token = 'AUTH_TOKEN'
service_sid = 'SERVICE_SID'
channel_sid = 'CHANNEL_SID'
message_sid = 'MESSAGE_SID'
@client = Twilio::REST::Client.new(account_sid, auth_token)

# Retrieve the message
service = @client.chat.v2.services(service_sid)
channel = service.channels(channel_sid) 
message = channel.messages(message_sid).fetch
puts "#{message.sid} #{message.body}"

File: ip-messaging/rest/messages/retrieve-messages/retrieve-messages.5.x.cs
// Download the twilio-csharp library from twilio.com/docs/libraries/csharp
using System;
using Twilio;
using Twilio.Rest.Chat.V2.Service.Channel;

class Example
{
    static void Main(string[] args)
    {
        // Find your Account SID and Auth Token at twilio.com/console
        const string accountSid = "accountSid";
        const string authToken = "authToken";
        const string serviceSid = "serviceSid";
        const string channelSid = "channelSid";
        const string messageSid = "messageSid";

        TwilioClient.Init(accountSid, authToken);

        // Retrieve a message
        var message = MessageResource.Fetch(serviceSid, channelSid, messageSid);
        Console.WriteLine(message.Body);
    }
}
